transport/dlr-malspp/transport/src/main/java/org/ccsds/moims/mo/testbed/util/sppimpl/util/SPPReader.java

```diff
@@ -147,25 +189,30 @@ public synchronized SpacePacket receive() throws IOException
 //        outPacket.setBody(trimmedBody);
     // Read CRC
     if (crcEnabled) {
-      int CRC = SPPHelper.computeCRC(inHeaderBuffer, data, outPacket.getOffset(), dataLength);
-
       is.read(inCrcBuffer);
+      if (!processedApids.inRange(apid)) {
```

Comment:
Current idea is to read the whole packet because otherwise the offsets get mixed up. That's why I read the whole packet first and then check if I need to process it. I can delete this redundant if, but this will result in CRC calculation for packets that I normally wouldn't need to process. Or maybe I'm interpreting your comment wrong. Could you state which lines you would put above/below which line?